### PR TITLE
[ASCII-1277] Remove sub packages from pkg/status. All agents now use the status component

### PR DESCRIPTION
**Pull request by GustavoCaso**
### What does this PR do?

After all the work done to migrate the different agent to use the status component:

- https://github.com/DataDog/datadog-agent/pull/23518
- https://github.com/DataDog/datadog-agent/pull/23472
- https://github.com/DataDog/datadog-agent/pull/23398
- https://github.com/DataDog/datadog-agent/pull/23263
- https://github.com/DataDog/datadog-agent/pull/22610
- https://github.com/DataDog/datadog-agent/pull/22115
- https://github.com/DataDog/datadog-agent/pull/22287

We can remove all the legacy subpakcges of `pkg/status` 🔥

### CODE REVIEW

File: cmd/agent/subcommands/run/command.go [hunk 1]
```diff
@@ -614,13 +613,7 @@ func startAgent(
...
 	go startDependentServices()
 
-	if err := otelcollector.Start(); err != nil {
-		return err
-	}
-	// TODO: (components) remove this once migrating the status package to components
-	otlpStatus.SetOtelCollector(otelcollector)
-
-	return nil
+	return otelcollector.Start()
```
Comment (louis-cqrl) [inline cmd/agent/subcommands/run/command.go:616]: `otelcollector.Start()` does return an error like the previous version ? or it can't fail anymore ?

Comment (GustavoCaso) [reply cmd/agent/subcommands/run/command.go:616]: I'm not entirely sure what you mean. 

Does `otelcollector.Start()` return an error? Yes

> previous version ?

Is there a previous version?

Comment (louis-cqrl) [reply cmd/agent/subcommands/run/command.go:616]: Okay my bad all okay, I thinks it is all good for me

**Review by louis-cqrl — APPROVED** [review]
LGTM

**Review by ogaca-dd — APPROVED** [review]

**Pull request merged**